lib/rspec/rails/matchers/send_email.rb

```diff
@@ -0,0 +1,125 @@
+# frozen_string_literal: true
+
+module RSpec
+  module Rails
+    module Matchers
+      # @api private
+      #
+      # Matcher class for `send_email`. Should not be instantiated directly.
+      #
+      # @see RSpec::Rails::Matchers#send_email
+      class SendEmail < RSpec::Rails::Matchers::BaseMatcher
+        INSPECT_EMAIL_ATTRIBUTES = %i[subject from to cc bcc].freeze
+
+        def initialize(criteria)
+          @criteria = criteria
+        end
+
+        # @private
+        def supports_value_expectations?
+          false
+        end
+
+        # @private
+        def supports_block_expectations?
+          true
+        end
+
+        def matches?(block)
+          define_matched_emails(block)
+
+          @matched_emails.one?
+        end
+
+        private
+
+        def diffable?
+          true
+        end
+
+        def match_when_negated(block)
+          define_matched_emails(block)
+          @matched_emails.empty?
+        end
+
+        def failure_message
+          result =
+            if multiple_match?
+              "More than 1 matching emails were sent."
+            else
+              "No matching emails were sent."
+            end
+          "#{result}#{sent_emails_message}"
+        end
+
+        def failure_message_when_negated
+          "Expected not to send an email but it was sent."
+        end
+
+        def deliveries
+          ActionMailer::Base.deliveries
+        end
+
+        def define_matched_emails(block)
+          before = deliveries.dup
+
+          block.call
+
+          after = deliveries
+
+          @diff = after - before
+          @matched_emails = select_matching(@diff)
+        end
+
+        def select_matching(emails)
+          @criteria.empty? ? emails : emails.select(&method(:matched_email?))
```

Comment:
`.empty?` check seems redundant
```ruby
[].all? { true } # => true
[].all? { false } # => true
```

```suggestion
          emails.select(&method(:matched_email?))
```